### PR TITLE
feat: implementation / enhancements of analytics  

### PR DESCRIPTION
**Pull request by saeedbashir**
This PR adds analytics to most parts of the app, as requested by the product team (@moiz994).

Note: @marcotuts We have discussed it internally, and I have renamed the already implemented analytics names for consistency between already implemented and newly implemented analytics, e.g  `Forgot_password_Clicked` to `Logistration_Forgot_Password_Clicked`, `Cookie_Policy_Clicked` to `Profile_Cookie_Policy_Clicked`.

### CODE REVIEW

**Comment by saeedbashir** [comment]
@volodymyr-chekyrta The tests are passing locally but failing on the PR, and I'm not getting any useful insights from the logs. Could you please look into it and see what needs to be fixed.

File: Profile/Profile/Presentation/DeleteAccount/DeleteAccountViewModel.swift [hunk 1]
```diff
@@ -27,26 +27,32 @@ public class DeleteAccountViewModel: ObservableObject {
     private let interactor: ProfileInteractorProtocol
     public let router: ProfileRouter
     public let connectivity: ConnectivityProtocol
+    let analytics: ProfileAnalytics
     
-    public init(interactor: ProfileInteractorProtocol, router: ProfileRouter, connectivity: ConnectivityProtocol) {
+    public init(interactor: ProfileInteractorProtocol, router: ProfileRouter, connectivity: ConnectivityProtocol, analytics: ProfileAnalytics) {
```
Comment (rnr) [inline Profile/Profile/Presentation/DeleteAccount/DeleteAccountViewModel.swift:32]: please fix line length warning

**Review by rnr — DISMISSED** [review]
LGTM
perhaps needs to coordinate with @shafqat-muneer and add analytics for Calendar sync too (maybe after his PR merging?)

File: Profile/Profile/Presentation/DeleteAccount/DeleteAccountViewModel.swift (hunk 1 above)
Comment (saeedbashir) [reply Profile/Profile/Presentation/DeleteAccount/DeleteAccountViewModel.swift]: Done

**Comment by saeedbashir** [comment]
> LGTM perhaps needs to coordinate with @shafqat-muneer and add analytics for Calendar sync too (maybe after his PR merging?)

We have created a separate task for calendar integration events.

**Review by rnr — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM 👍

**Pull request merged**